lib/mongoid/extensions/hash.rb

```diff
@@ -198,7 +198,7 @@ def to_criteria
       def value_for(operator, klass, key, value)
         case operator
         when "$rename" then value.to_s
-        when "$addToSet", "$push" then value.mongoize
+        when "$addToSet", "$push", '$pull', '$pop' then value.mongoize
```

Comment:
`$pullAll`?

Reply:
Thanks @johnnyshields. I looked close at this exact question when I was evaluating these PRs, and it's not necessary to specify `$pullAll` here. In the case of `$push`, `$pull`, etc, the value of the operation may different than the declared value of the field itself (because the field is array-valued), and so must be handled specially. For `$pullAll`, the value is an array, for an array-valued field, and so it is handled fine by the default case.

Reply:
@jamis any chances to see this changes in 8.1-stable and next release 8.1.x?

Reply:
Thanks for the nudge, @dem. I'm not sure how this fell through the cracks, but I'll get it merged today.